### PR TITLE
Announcing fastai integration

### PR DESCRIPTION
**Pull request by omarespejel**
🚀 Blog announcing the integration with fastai. [Here is a brief intro](https://colab.research.google.com/gist/omarespejel/9ba054ce74c7a4d4408085b611124bdf) using Colab.

I want to show:
- Why this integration is great for open ML (and the fastai and Hugging Face missions).
- Why sharing to the Hub is desired.
- How to install the integration.
- Sharing Learners with the Hub.
- Loading Learners with the Hub.
- What is next? (Join HF discord, take the fastai's courses, load Learners...)

Any feedback would be highly appreciated 🤗

Tagging @muellerzr and @sgugger for their huge fastai experience 🔥.

### CODE REVIEW

File: fastai.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+fastai is an [open-source Deep Learning library](https://github.com/fastai/fastai) that leverages PyTorch and Python to provide high-level components. It quickly delivers state-of-the-art output on text, vision, and tabular data. However, fastai is more than just a library; it's an entire ecosystem of notable contributors. As some examples, check out their [book](https://github.com/fastai/fastbook), [documentation](https://docs.fast.ai/), and [courses](https://course.fast.ai/). Join the fastai [Discord](https://discord.com/invite/YKrxeNn) and [forums](https://forums.fast.ai/). It is a guarantee that you will learn by engaging with their community.
+
+Because of all this, and more (the writer of this post started his journey thanks to the fastai course), we are proud to announce that Fastai practitioners can now share and upload models to Hugging Face Hub with a single line of Python.
```
Comment (sgugger) [inline fastai.md:33]:
```suggestion
Because of all this, and more (the writer of this post started his journey thanks to the fastai course), we are proud to announce that fastai practitioners can now share and upload models to Hugging Face Hub with a single line of Python.
```

File: fastai.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+ 👉 In this post, we will quickly introduce the integration between Fastai and Hub. Check out [this Google Colab notebook](https://colab.research.google.com/gist/omarespejel/9ba054ce74c7a4d4408085b611124bdf) for a hands-on introduction.
```
Comment (sgugger) [inline fastai.md:35]:
```suggestion
 👉 In this post, we will quickly introduce the integration between fastai and Hub. Check out [this Google Colab notebook](https://colab.research.google.com/gist/omarespejel/9ba054ce74c7a4d4408085b611124bdf) for a hands-on introduction.
```

File: fastai.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+## Why share to the Hub?
+
+The Hub is a central platform where anyone can share and explore models, datasets, and ML demos. It aims to build the most extensive collection of Open Source models, datasets, and demos.
+
+Sharing on the Hub could amplify the impact of a fastai model by making it available for others to download and explore. Also, fastai models take great advantage of transfer learning; load someone else's model as the basis for your task.
+
+Anyone can access all the fastai models in the Hub by filtering the [huggingface.co/models](https://huggingface.co/models?library=fastai&sort=downloads) webpage by the fastai library, as in the image below.
+
+To know more about the Hub, refer to this introduction.
```
Comment (sgugger) [inline fastai.md:48]: Should this introduction be a link to somewhere?

File: fastai.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+To share models in the Hub, you will need to have a user. Create it on the [Hugging Face website](https://huggingface.co/).
+
+## Sharing and finding Learners to the Hub
+
+There are three options to login to the Hugging Face Hub; your token will be available in your Account Settings:
+1. Type `huggingface-cli login` in your terminal and enter your token.
+
+2. If in a python notebook, you can use `notebook_login`.
+
+```
```
Comment (sgugger) [inline fastai.md:81]:
```suggestion
```py
```

File: fastai.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+from huggingface_hub import notebook_login
+notebook_login()
+```
+
+3. Use the `token` argument of the `push_to_hub_fastai` function.
+
+Input `push_to_hub_fastai` (refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai)) with the Learner you want to upload and the repository id for the Hub in the format of "namespace/repo_name". The namespace can be an individual account or an organization you have write access to (for example, 'fastai/stanza-de'). For more details, refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai).
+
+```
```
Comment (sgugger) [inline fastai.md:90]:
```suggestion
```py
```

File: fastai.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+2. If in a python notebook, you can use `notebook_login`.
+
+```
+from huggingface_hub import notebook_login
+notebook_login()
+```
+
+3. Use the `token` argument of the `push_to_hub_fastai` function.
+
+Input `push_to_hub_fastai` (refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai)) with the Learner you want to upload and the repository id for the Hub in the format of "namespace/repo_name". The namespace can be an individual account or an organization you have write access to (for example, 'fastai/stanza-de'). For more details, refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai).
```
Comment (sgugger) [inline fastai.md:88]:
```suggestion
Input `push_to_hub_fastai` (refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai)) with the `Learner` you want to upload and the repository id for the Hub in the format of "namespace/repo_name". The namespace can be an individual account or an organization you have write access to (for example, 'fastai/stanza-de'). For more details, refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai).
```

**Review by sgugger — APPROVED** [review]
Thanks for writing this! Just for using the right terms:
- fastai is the library (no capital): the fastai library
- fast.ai is the company/community/ecocystem: the fast.ai course/book etc.

File: _blog.yml [hunk 7]
```diff
@@ -713,3 +713,15 @@
...
+  title: "Welcome fastai to the Hugging Face Hub"
+  author: espejelomar
+  thumbnail: /blog/assets/64_fastai/fastai_hf_blog.png
+  date: May 5, 2022
+  tags:
+  - guide
+  - open-source-collab
+  - community
+  - nlp
+  - cv
```
Comment (osanseviero) [inline _blog.yml:727]: I would remove these tags since this blog post is not super specific about these domains, it's more about a community thing or open source collaboration

```suggestion
```

File: fastai.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        <div class="bfc">
+            <code>espejelomar</code>
+            <span class="fullname">Omar Espejel</span>
+        </div>
+    </a>
+</div>
+
+## Making neural nets uncool again... and sharing them
+
+Few have done as much as the [fastai](https://www.fast.ai/) ecosystem to make Deep Learning accessible. Our mission at Hugging Face is to democratize good Machine Learning. Let's make exclusivity in access to Machine Learning, including [pre-trained models](https://huggingface.co/models), a thing of the past and let's push this amazing field even further.
```
Comment (osanseviero) [inline fastai.md:29]: nit: fast.ai since you're talking about the community/ecosystem

```suggestion
Few have done as much as the [fast.ai](https://www.fast.ai/) ecosystem to make Deep Learning accessible. Our mission at Hugging Face is to democratize good Machine Learning. Let's make exclusivity in access to Machine Learning, including [pre-trained models](https://huggingface.co/models), a thing of the past and let's push this amazing field even further.
```

File: fastai.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+fastai is an [open-source Deep Learning library](https://github.com/fastai/fastai) that leverages PyTorch and Python to provide high-level components. It quickly delivers state-of-the-art output on text, vision, and tabular data. However, fastai is more than just a library; it is maintained by [fast.ai](https://www.fast.ai/), the company, and has grown into an entire ecosystem of notable contributors. As some examples, check out fast.ai's [book](https://github.com/fastai/fastbook) and [courses](https://course.fast.ai/). Join the fast.ai [Discord](https://discord.com/invite/YKrxeNn) and [forums](https://forums.fast.ai/). It is a guarantee that you will learn by engaging with their community.
```
Comment (osanseviero) [inline fastai.md:31]:
```suggestion
fastai is an [open-source Deep Learning library](https://github.com/fastai/fastai) that leverages PyTorch and Python to provide high-level components to train fast and accurate neural networks with state-of-the-art output on text, vision, and tabular data. However, fastai is more than just a library; it has grown into a thriving ecosystem of open source contributors and people learning about neural networks. As some examples, check out their [book](https://github.com/fastai/fastbook) and [courses](https://course.fast.ai/). Join the fast.ai [Discord](https://discord.com/invite/YKrxeNn) and [forums](https://forums.fast.ai/). It is a guarantee that you will learn by being part of their community!
```

File: fastai.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+Because of all this, and more (the writer of this post started his journey thanks to the fast.ai course), we are proud to announce that fastai practitioners can now share and upload models to Hugging Face Hub with a single line of Python.
+
+ 👉 In this post, we will quickly introduce the integration between fastai and Hub. Check out [this Google Colab notebook](https://colab.research.google.com/gist/omarespejel/9ba054ce74c7a4d4408085b611124bdf) for a hands-on introduction.
```
Comment (osanseviero) [inline fastai.md:35]: You can add the notebook to https://github.com/huggingface/blog/tree/main/notebooks

Comment (osanseviero) [inline fastai.md:35]:
```suggestion
Because of all this, and more (the writer of this post started his journey thanks to the fast.ai course), we are proud to announce that fastai practitioners can now share and upload models to Hugging Face Hub with a single line of Python.

 👉 In this post, we will introduce the integration between fastai and the Hub. Check out [this Google Colab notebook](https://colab.research.google.com/gist/omarespejel/9ba054ce74c7a4d4408085b611124bdf) for a hands-on introduction.
```

File: fastai.md [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+
+*We want to thank the fast.ai community, notably Jeremy Howard, for their feedback 🤗. This blog is heavily inspired by the [Hugging Face Hub section](https://docs.fast.ai/huggingface.html) in the fastai docs.
```
Comment (osanseviero) [inline fastai.md:37]:
```suggestion
* We want to thank the fast.ai community, notably Jeremy Howard, for their feedback 🤗. This blog is heavily inspired by the [Hugging Face Hub section](https://docs.fast.ai/huggingface.html) in the fastai docs.
```

File: fastai.md [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+
+
+## Why share to the Hub?
+
+The Hub is a central platform where anyone can share and explore models, datasets, and ML demos. It aims to build the most extensive collection of Open Source models, datasets, and demos.
```
Comment (osanseviero) [inline fastai.md:42]:
```suggestion
The Hub is a central platform where anyone can share and explore models, datasets, and ML demos. It has the most extensive collection of Open Source models, datasets, and demos.
```

File: fastai.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+
+Sharing on the Hub could amplify the impact of a fastai model by making it available for others to download and explore. Also, fastai models take great advantage of transfer learning; load someone else's model as the basis for your task.
```
Comment (osanseviero) [inline fastai.md:44]:
```suggestion
Sharing on the Hub amplifies the impact of your fastai models by making them available for others to download and explore. You can also use transfer learning with fastai models; load someone else's model as the basis for your task.
```

File: fastai.md [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+
+Anyone can access all the fastai models in the Hub by filtering the [huggingface.co/models](https://huggingface.co/models?library=fastai&sort=downloads) webpage by the fastai library, as in the image below.
```
Comment (osanseviero) [inline fastai.md:46]:
```suggestion
Anyone can access all the fastai models in the Hub by filtering the [hf.co/models](https://huggingface.co/models?library=fastai&sort=downloads) webpage by the fastai library, as in the image below.
```

File: fastai.md [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+
+## Installation
+
+Install `huggingface_hub`. Additionally, the integration functions require the following packages:
+
+- toml,
+- fastai>=2.4,
+- fastcore>=1.3.27
+
+You can install these packages manually or specify ["fastai"] when installing `huggingface_hub`, and your environment will be ready:
```
Comment (osanseviero) [inline fastai.md:61]:
```suggestion
The `huggingface_hub` library is a lightweight Python client with utility functions to interact with the Hugging Face Hub. In order to push fastai models to the Hub, you require to have some libraries pre-installed (fastai, fastcore, toml), but you can install them automatically by specifying ["fastai"] when installing `huggingface_hub`, and your environment will be ready:
```

File: fastai.md [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+2. If in a python notebook, you can use `notebook_login`.
+
+```py
+from huggingface_hub import notebook_login
+notebook_login()
+```
+
+3. Use the `token` argument of the `push_to_hub_fastai` function.
+
+Input `push_to_hub_fastai` (refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai)) with the `Learner` you want to upload and the repository id for the Hub in the format of "namespace/repo_name". The namespace can be an individual account or an organization you have write access to (for example, 'fastai/stanza-de'). For more details, refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai).
```
Comment (osanseviero) [inline fastai.md:90]:
```suggestion
You can input `push_to_hub_fastai` with the `Learner` you want to upload and the repository id for the Hub in the format of "namespace/repo_name". The namespace can be an individual account or an organization you have write access to (for example, 'fastai/stanza-de'). For more details, refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai).
```

File: fastai.md [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+```bash
+pip install git+https://github.com/huggingface/huggingface_hub#egg=huggingface-hub["fastai"]
+```
+To share models in the Hub, you will need to have a user. Create it on the [Hugging Face website](https://huggingface.co/).
+
+## Sharing a `Learner` to the Hub
+
+*A [`Learner` is a fastai object](https://docs.fast.ai/learner.html#Learner) that bundles a model, data loaders, and a loss function. We will use the words `Learner` and Model interchangeably throughout this post.
+
+There are three options to login to the Hugging Face Hub; your token will be available in your Account Settings:
```
Comment (osanseviero) [inline fastai.md:78]: Maybe mention users that they need to create a write token in http://hf.co/settings/tokens before any of these steps

File: fastai.md [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
+from huggingface_hub import notebook_login
+notebook_login()
+```
+
+3. Use the `token` argument of the `push_to_hub_fastai` function.
+
+Input `push_to_hub_fastai` (refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai)) with the `Learner` you want to upload and the repository id for the Hub in the format of "namespace/repo_name". The namespace can be an individual account or an organization you have write access to (for example, 'fastai/stanza-de'). For more details, refer to the [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.push_to_hub_fastai).
+
+```py
+from huggingface_hub import push_to_hub_fastai
```
Comment (osanseviero) [inline fastai.md:93]: As discussed internally, maybe let's add the code for the learner here and in the colab

File: fastai.md [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
+
+![Fastai Model Card](assets/64_fastai/hf_model_card.png)
+
+## Loading a `Learner` from Hub
+
+To load a model in the Hub with the id `ITESM/fastai_model`:
+
+```py
+from huggingface_hub import from_pretrained_fastai
+learner = from_pretrained_fastai("ITESM/fastai_model")
```
Comment (osanseviero) [inline fastai.md:109]: I think it might be nicer if you load the same model you pushed and then do an inference with the image classifier for example

File: fastai.md [hunk 20]
```diff
@@ -0,0 +1,124 @@
...
+```
+
+The [Hub Client documentation](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/mixins#huggingface_hub.from_pretrained_fastai) includes addtional details on `from_pretrained_fastai`.
+
+
+## What is next?
```
Comment (osanseviero) [inline fastai.md:115]:
```suggestion
## What's next?
```

File: fastai.md [hunk 21]
```diff
@@ -0,0 +1,124 @@
...
+
+## Why share to the Hub?
+
+The Hub is a central platform where anyone can share and explore models, datasets, and ML demos. It aims to build the most extensive collection of Open Source models, datasets, and demos.
+
+Sharing on the Hub could amplify the impact of a fastai model by making it available for others to download and explore. Also, fastai models take great advantage of transfer learning; load someone else's model as the basis for your task.
+
+Anyone can access all the fastai models in the Hub by filtering the [huggingface.co/models](https://huggingface.co/models?library=fastai&sort=downloads) webpage by the fastai library, as in the image below.
+
+To know more about the Hub, refer to [this introduction](https://github.com/huggingface/education-toolkit/blob/main/01_huggingface-hub-tour.md).
```
Comment (osanseviero) [inline fastai.md:48]: I would add some of the benefits from https://huggingface.co/docs/hub/adding-a-library#integrate-your-library-with-the-hub

* Free model hosting
* Built-in file versioning, even for large files
* Model cards

**Review by osanseviero — COMMENTED** [review]
Thank you for this blog post! This is great :rocket: 

I left some suggestions on wording and some ideas

File: fastai.md (hunk 21 above)
Comment (omarespejel) [reply fastai.md:48]: Great! I wrote:
```
In addition to free model hosting and exposure to the broader community, the Hub has built-in [version control based on git and git-lfs](https://huggingface.co/docs/transformers/model_sharing#repository-features) (for large files) and [model cards](https://huggingface.co/docs/hub/model-repos#what-are-model-cards-and-why-are-they-useful) for discoverability and reproducibility. For more information on navigating the Hub, see [this introduction](https://github.com/huggingface/education-toolkit/blob/main/01_huggingface-hub-tour.md).
```

**Comment by omarespejel** [comment]
@osanseviero  thank you for your great feedback! It was very valuable. Now the blog is on another level 🚀.

**Review by osanseviero — APPROVED** [review]
Thanks for this PR :rocket: this looks great :fire: 

Feel free to merge when you want
